### PR TITLE
Updates public API specs.

### PR DESCRIPTION
**Pull request by cloudify**
Adds more documentation about messages and the verification flow

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please include a Pivotal story at the beginning of the PR title (see below).
  </td>
    </tr>
  </tbody>
</table>



  Example of PR titles that include pivotal stories:

  * single story: `[#123456] my PR title`
  * multiple stories: `[#123456,#123457,#123458] my PR title`

    
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: api/public_api_v1.yaml [hunk 1]
```diff
@@ -1,8 +1,101 @@
...
+    Digital Citizenship platform of the Italian Government.
+    The platform enables services to communicate with Italian Citizens via
+    the [IO app](https://io.italia.it/).
+
+    # How to get an API key
+
+    To get access to this API, you'll need to register on the
+    [Digital Citizenship Developer Portal](https://developer.cd.italia.it/).
+
+    Once registered, you will receive an API Key (`Ocp-Apim-Subscription-Key`)
```
Comment (gunzip) [inline api/public_api_v1.yaml]: note that after registration, you have to click 'subscribe to the digital citizenship api' button

File: api/public_api_v1.yaml [hunk 2]
```diff
@@ -1,8 +1,101 @@
...
+      * A required `subject`: a short description of the topic.
+      * A required `markdown` body: a Markdown representation of the body (see
+        below on what Markdown tags are allowed).
+      * An optional `payment_data`: in case the message is a payment request,
+        the _payment data_ will enable the recipient to pay the requested amount
+        via [PagoPA](https://www.agid.gov.it/it/piattaforme/pagopa).
+
+    ## Allowed Markdown formatting
+
+    Not all Markdown formatting is currently available. Currently you can use
```
Comment (gunzip) [inline api/public_api_v1.yaml]: atm there's no such blacklist / whitelist

File: api/public_api_v1.yaml [hunk 3]
```diff
@@ -1,8 +1,101 @@
...
+    the following formatting:
+
+      * Headings
+      * Text stylings (bold, italic, etc...)
+      * Lists (bullet and numbered)
+
+    ## Process for sending a message to a citizen
+
+    Not every citizen will be interested in what you have to say and not every
+    citizen you want to communicate to will be registed on the Digital
```
Comment (gunzip) [inline api/public_api_v1.yaml]: the will

File: api/public_api_v1.yaml (hunk 2 above)
Comment (cloudify) [reply api/public_api_v1.yaml]: yeah I know, but they don't :D

**Pull request merged**